### ISSUE TITLE
@RequestScope is missing when using NettyClientCustomizer

### ISSUE THREAD

**Issue by s-af** [micronaut-projects/micronaut-core#10116]
### Expected Behavior

I am trying to integrate Logbook for client's and server's request/responses based on examples in documentation https://docs.micronaut.io/latest/guide/#nettyClientPipeline. Since major [bug](https://github.com/zalando/logbook/issues/1216) in Logbook fixed it seems to be a valid case (perhaps you can remove mention of the bug from the documentation). 
The issues start when I try to customize LogbookClientHandler and use condition based on (a)RequestScope bean value:
```
    channel.pipeline().addBefore(
            ChannelPipelineCustomizer.HANDLER_MICRONAUT_HTTP_RESPONSE,
            "logbook",
            new LogbookClientHandler(
                    Logbook.builder()
                            .condition(httpRequest -> requestContext.getMagicUser())
                            .build()
            )
    );
```
In described case I am getting error:
```
java.lang.IllegalStateException: No request present
```
Netty**Server**Customizer and/or custom ChannelHandlers behave in same way.

Expected Behavior: (a)RequestScope work in combination with NettyClientCustomizer and NettyServerCustomizer
 

### Actual Behaviour

(a)RequestScope is not working in combination with NettyClientCustomizer and NettyServerCustomizer

### Steps To Reproduce

Retrieve RequestScoped value inside netty channel handler.

Reproducible example can be found [here](https://github.com/s-af/customizer-with-context.git).

### Environment Information

- Win10
- JDK17

### Example Application

https://github.com/s-af/customizer-with-context.git

### Version

4.1.6

**Comment by yawkat** [comment]
I'm sorry but I don't believe this is feasible. Netty pipeline handlers happen outside Micronaut HTTP request processing. We cannot determine the current request in this context eg because of pipelining.

**Comment by s-af** [comment]
@yawkat, I can agree on it when we speak about Netty**Server**Customizer. But when we speak about Netty**Client**Customizer it seems for me a valid case. Client calls happen inside Micronaut HTTP request processing.

**Comment by yawkat** [comment]
Fair enough, there is a unique association between the channel and the request in the client. However it's still not feasible to support your test case because there is no such unique association between the event loop thread and the request, so we cannot rely on thread locals for this use case. The only option is to pass the channel to the framework somehow. But I will leave this open.

**Comment by s-af** [comment]
Thank you! 
It would be really helpful if you able to make work ClientCustomizer. In ServerCustomizer I have access to the original http request, so will be able to analyze it somehow. But inside ClientCustomizer there is no options how to get information about original request.

**Comment by graemerocher** [comment]
This is a feature request for request context propagation into Netty customizers, not a conventional bean-resolution bug. A request-to-channel association exists, but arbitrary handlers on a shared Netty event loop do not have a reliable unique thread-to-request mapping, so a thread-local `RequestScope` cannot be inferred there. A durable solution would need an explicit request/channel context API or lifecycle hook exposed to `NettyClientCustomizer`/server customizers, with cleanup at request completion and clear behavior for multiplexed HTTP/2; the existing reproducer is useful for designing that API.
